components/feature/media/src/test/java/mozilla/components/feature/media/service/MediaSessionServiceDelegateTest.kt

```diff
@@ -238,4 +245,27 @@ class MediaSessionServiceDelegateTest {
         mediaSessionCallback.onPlay()
         verify(controller).play()
     }
+
+    @Test
+    fun `destroying service will stop all playback sessions`() {
+        val controller: MediaSession.Controller = mock()
+        val initialState = BrowserState(
+            tabs = listOf(createTab(
+                "https://www.mozilla.org",
+                mediaSessionState = MediaSessionState(
+                    controller,
+                    playbackState = MediaSession.PlaybackState.PLAYING
+                )
+            ))
+        )
+        val store = BrowserStore(initialState)
+        val service: AbstractMediaSessionService = mock()
+        val delegate = spy(MediaSessionServiceDelegate(testContext, service, store))
+
+        delegate.onCreate()
+
+        delegate.onDestroy()
+        verify(service, never()).stopSelf()
+        verify(delegate).destroy()
```

Comment:
I wonder if we could test that we call things like `abandon()` on the audio focus (what we want to do here) instead of `destroy()` (which can change behavior over time). Like if I remove the `abandon()` call then this test still passes.

Reply:
As agreed, I'll create an issue to explore if we can do a better test for `audiofocus.abandon`.

Reply:
Issue created https://github.com/mozilla-mobile/android-components/issues/9260